src/community/assortativity.jl

```diff
@@ -34,18 +48,33 @@ function assortativity(g::AbstractGraph{T}) where T
     return assortativity_coefficient(g, sjk, sj, sk, sjs, sks, nue)
 end
 
+function assortativity(g::AbstractGraph{T},attributes::Vector{N}) where {T,N<:Number}
+    nue  = ne(g)
+    sjk = sj = sk = sjs = sks = zero(N)
+    for d in edges(g)
+        j = attributes[src(d)]
+        k = attributes[dst(d)]
+        sjk += j*k
+        sj  += j
+        sk  += k
+        sjs += j^2
```

Comment:
under what circumstances will `sjs` and `sks` overflow? If `N` is `Int8`, for example, and one of the elements in attributes is `7`, then you've got a potential problem.